### _Should I calculate gene expression using constitutive or all mRNA aligning probests?_ ###

_**Answer**_: This is a common question for new users. While the development team currently recommends the use of just constitutive probesets for gene expression calculation, users are encouraged to try both and compare differences with the two methods. As of 2010, we have not performed any careful studies to assess which method has the greatest sensitivity and specificity. However, we anticipate both methods have pros and cons.

For example, focusing on constitutive probesets positively biases toward exons that do not demonstrate alternative expression in the major mRNA databases. However, given that many exons in the genome are alternative expressed with novel patterns using RNA sequencing analyses, such an assumption may negatively impact alternative exon detection. Alternatively, if many exons within a gene are alternatively expressed (e.g., alternative promoter), then too much noise may exist to assess alternative expression in a specific manner.

Methods such as [MADS](http://www.rnajournal.org/cgi/pmidlookup?view=long&pmid=18566192) provide an alternative approach, which relies on prediction of constitutive exons based on multiple array datasets.