# Professional Biosketch - Kristina Hanspers #

![http://www.gladstone.ucsf.edu/gladstone/files/conklin/Hanspers.jpg](http://www.gladstone.ucsf.edu/gladstone/files/conklin/Hanspers.jpg)

### Education ###

2000 Master of Engineering (Biotechnology), Uppsala University School of Engineering, Uppsala, Sweden

### Experience ###
4/2003 – present Senior Research Associate, Conklin laboratory, J. David Gladstone Institutes, San Francisco, CA

10/2000 – 4/2003 Research Associate, Genomics Core, J. David Gladstone Institutes, San Francisco, CA

3/1999 – 9/1999 Graduate Student, Planet Biotechnology Inc., Mountain View, CA

### Scientific Publications ###

van Iersel MP, Pico AR, Kelder T, Gao J, Ho I, Hanspers K, Conklin BR,
Evelo CT. The BridgeDb framework: standardized access to gene, protein and metabolite identifier mapping services. BMC Bioinformatics. 2010 Jan 4;11:5.

Salomonis N, Nelson B, Vranizan K, Pico AR, Hanspers K, Kuchinsky A, Ta L, Mercola M,
Conklin BR. PLoS Comput Biol. 2009 Nov;5(11):e1000553. Alternative splicing in the differentiation of human embryonic stem cells into cardiac precursors.

Kelder T, Pico AR, Hanspers K, van Iersel MP, Evelo C, Conklin BR. Mining biological pathways using WikiPathways web services. PLoS One. 2009 Jul 30;4(7):e6447.

van Iersel MP, Kelder T, Pico AR, Hanspers K, Coort S, Conklin BR, Evelo C. Presenting and exploring biological pathways with PathVisio. BMC Bioinformatics. 2008 Sep 25;9:399.

Pico AR, Kelder T, van Iersel MP, Hanspers K, Conklin BR, Evelo C. PLoS Biol. WikiPathways: pathway editing for the people. 2008 Jul 22;6(7):e184.

Nervall M, Hanspers P, Carlsson J, Boukharta L, Aqvist J. Predicting binding modes from free energy calculations. J Med Chem. 2008 May 8;51(9):2657-67

Cline MS, Smoot M, Cerami E, Kuchinsky A, Landys N, Workman C, Christmas R, Avila-Campilo I, Creech M, Gross B, Hanspers K, Isserlin R, Kelley R, Killcoyne S, Lotia S, Maere S, Morris J, Ono K, Pavlovic V, Pico AR, Vailaya A, Wang PL, Adler A, Conklin BR, Hood L, Kuiper M, Sander C, Schmulevich I, Schwikowski B, Warner GJ, Ideker T, Bader GD. Nat Protoc. Integration of biological networks and gene expression data using Cytoscape. 2007;2(10):2366-82.

Nord AS, Vranizan K, Tingley W, Zambon AC, Hanspers K, Fong LG, Hu Y, Bacchetti P,
Ferrin TE, Babbitt PC, Doniger SW, Skarnes WC, Young SG, Conklin BR. Modeling insertional mutagenesis using gene length and expression in murine embryonic stem
cells. PLoS One. 2007 Jul 18;2(7):e617.

Salomonis N, Hanspers K, Zambon AC, Vranizan K, Lawlor SC, Dahlquist KD, Doniger SW, Stuart J, Conklin BR, Pico AR. GenMAPP 2: new features and resources for pathway analysis. BMC Bioinformatics. 2007 Jun 24;8:217.

Yu X, Griffith WC, Hanspers K, Dillman JF 3rd, Ong H, Vredevoogd MA, Faustman EM. A system-based approach to interpret dose- and time-dependent microarray data: quantitative integration of gene ontology analysis for risk assessment. Toxicol Sci. 2006 Aug;92(2):560-77.

Barker CS, Griffin C, Dolganov GM, Hanspers K, Yang JY, Erle DJ. Increased DNA microarray hybridization specificity using sscDNA targets. BMC Genomics. 2005 Apr 22;6(1):57.

Lee MS, Hanspers K, Barker CS, Korn AP, McCune JM. Gene expression profiles during human CD4+ T cell differentiation. Int Immunol. 2004 Aug;16(8):1109-24.

Barczak A, Rodriguez MW, Hanspers K, Koth LL, Tai YC, Bolstad BM, Speed TP, Erle DJ. Spotted long oligonucleotide arrays for human gene expression analysis. Genome Res. 2003
Jul;13(7):1775-85.

Buschard K, Hanspers K, Fredman P, Reich EP. Treatment with sulfatide or its precursor, galactosylceramide, prevents diabetes in NOD mice. Autoimmunity. 2001;34(1):9-17.