### _How does a pair-wise alternative exon analysis differ from multiple group comparisons?_ ###

_**Answer**_: Statistics for a pair-wise comparison are calculated by comparing the expression values for two biological groups. For exon array analyses (splicing-index or FIRMA), these means comparison of the gene expression corrected intensities (or residuals) in the control and experimental groups to derive a alternative exon fold change and p-value based on those two groups. For a junction array analysis, this also relies on two groups, control and experimental, but with different algorithm options (ASPIRE and linear regression), involving multiple probesets.

When more than two biological sample groups are analyzed for alternative exon expression, AltAnalyze compares the two groups with the most extreme expression values. In the case of an exon array, these are the two groups with the smallest and lowest normalized intensities (mean) to derive a score (splicing-index or FIRMA). To derive a p-value, the variance in all groups are examined by MiDAS and/or by a f-test of the normalized intensities all of group samples.