# Professional Biosketch - Nathan Salomonis #

![http://genmapp.org/graphics/nathan_salomonis.jpg](http://genmapp.org/graphics/nathan_salomonis.jpg)


  * [professional website](http://pinscher.ucsf.edu/~nsalomonis/informatics/)
  * http://milosalomonis.org
  * [Linkedin](http://www.linkedin.com/pub/nathan-salomonis/4/577/20a)

### EDUCATION/TRAINING ###

|**INSTITUTION AND LOCATION**|**DEGREE**|**YEARS**|**MAJOR**|
|:---------------------------|:---------|:--------|:--------|
|University of California, Los Angeles, CA|B.S.      |1994–1998|Neuroscience|
|University of California, San Francisco, CA|Ph.D      |2002–2008|Pharmacogenomics|
|Gladstone Institute of Cardiovascular Disease|Postdoctoral|2008–    |Bioinformatics|

### CURRENT POSITION ###

Gladstone: Postdoctoral Fellow

### PERSONAL STATEMENT ###

The emergence of high-throughput genomic methods has sparked a revolution in the development of innovative bioinformatics approaches. While some of these approaches focus on improved algorithms, others focus on combining seemingly disparate data sources in meaningful ways. My career has focused primarily on the latter—developing new workflows for novel biological discovery. In the late 90s and early 2000s, this work focused on the development of new visualization methods for gene expression data along biological pathways (GenMAPP) and applying these techniques to large disease and developmental datasets for target discovery. In recent years, I have focused on the discovery of novel functional differences between alternative isoforms, using existing protein and binding site annotation databases. I believe these and expanded methods will prove to be critical in understanding the biological implications of second and third generation sequencing experiments, as they have from exon-tiling array experiments.

### POSITIONS AND HONORS ###
Gladstone Institute of Cardiovascular Disease, UCSF	Postdoctoral Fellow
Principle Investigator: Bruce Conklin, MD	2008–
  * Lead developer of AltAnalyze: software design and implementation; database design and implementation; grant writing (awarded NIH RO1 supplement); documentation, tutorials, help-desk and Google groups administrator; Google summer of code mentor; primary author of related publications.
  * Lead developer of GO-Elite: software design and implementation; database design and implementation; documentation, tutorials, help-desk and Google groups administrator.
  * Associate developer of DomainGraph: software design; tutorials; co-primary author of related publication.
  * microRNA bioinformatics: developed new methods for analysis of microRNA binding site heptamer density and enrichment from column-based microRNA target assays (microarray), in collaboration with the Deepak Srivastava lab.

Grants and Fellowships Awarded:
  * NIH ARRA Supplement, 2009–2011

University of California, San Francisco	Graduate Student
Principle Investigator: Bruce Conklin, MD	2002–2008
  * Stem cell biology: examined whole-genome differences in exon and exon-junction expression between embryonic stem (ES) cells and differentiated fates; RNA-sample preparation; ES cell culture and differentiation; selective mRNA isoform RNA inhibition; microRNA reporter assays; recombinant DNA technologies; quantitative PCR and mRNA isoform RT-PCR validation.
  * Alternative splicing bioinformatics: developed new software (AltAnalyze-alpha version) for the analysis of exon and exon-junction tiling microarray datasets to predict alternative splicing and alternative promoter selection, predict removal of microRNA binding sites and predict modification of protein domains (InterPro) and motifs (UniProt) due to alternative transcript regulation.
  * Pathway analysis bioinformatics: developed methods for Gene Ontology hierarchy pruning and WikiPathway-associated pathway analysis (GO-Elite). Established a base of users for testing and further development. Featured analysis method in several publications by outside laboratories. Design and grant writing for GenMAPP-CS (awarded NIH RO1).
  * Cancer prognostic bioinformatics: developed new software for breast cancer array classification using orthology to murine mammary gland developmental markers and agglomerative/partioned clustering (OncoSplit).
  * Transcriptional regulation bioinformatics: developed software to identify global exonic, intronic and genic differences with disruption of transcriptional elongation factors in zebrafish.

Grants and Fellowships Awarded:
  * Achievement Rewards for College Scientists (ARCS) Scholarship 2005–2006
  * Ira Herskowitz Award for Excellence in PSPG Graduate Program 2005
  * NIH Training Grant (T32 GM07175) 2003–2004

Gladstone Institute of Cardiovascular Disease, UCSF	Research Associate
Principle Investigator: Bruce Conklin, MD	1998–2002
  * Uterine biology: examined whole-genome differences in gene expression between time-points of uterine gestation; microarray analysis methods; expression clustering; associated GenMAPP pathway development; development of chromosomal expression clustering software (Python).
  * Molecular biology: used standard recombinant DNA technologies and protein immunohistochemistry to modify and validate the function of G-protein coupled receptors and G-alpha protein subunits.
  * Pathway analysis bioinformatics: principle feature design of the pathway visualization software GenMAPP version 1.0, 2.0 and MAPPFinder; designed and implemented GenMAPP database; developed pathway content for GenMAPP; documentation, tutorials and help-desk.
  * Web developer: developed http://www.genmapp.org and educational web-content.

Awards:
  * Gladstone Institute of Cardiovascular Disease, 2001 Annual Award of Excellence
  * Gladstone Institute of Cardiovascular Disease Travel Award, 2000

University of California, Los Angeles	Student Researcher
Principle Investigator: Jeffery Twiss, MD	1997–1998
  * Neurobiology: examined the role of pro-inflammatory cytokine production on CNS glial cells, using immunohistochemistry, cell culture and cell survival assays.

### SELECTED PUBLICATIONS ###
**Salomonis N**, **Emig D**, Baumbach J, Lengauer T, Conklin BR, Albrecht M. (2010) AltAnalyze and DomainGraph: analyzing and visualizing exon expression data. Nucleic Acids Res. 1;38. **co-first authors**.

Salomonis N, Schlieve CR, Pereira L, Wahlquist C, Colas A, Zambon AC, Vranizan K, Spindler MJ, Pico AR, Cline MS, Clark TA, Williams A, Blume JE, Samal E, Mercola M, Merrill BJ, Conklin BR. (2010) Alternative splicing regulates mouse embryonic stem cell pluripotency and differentiation. Proc Natl Acad Sci U S A. 107(23):10514-9.

Nakamura K, Salomonis N, Tomoda K, Yamanaka S, Conklin BR. (2009) G(i)-coupled GPCR signaling controls the formation and organization of human pluripotent colonies. PLoS One. 10;4(11):e7780.

Salomonis N, Nelson B, Vranizan K, Pico AR, Hanspers K, Kuchinsky A, Ta L, Mercola M, Conklin BR. (2009) Alternative splicing in the differentiation of human embryonic stem cells into cardiac precursors. PLoS Comput Biol. 5(11):e1000553.

Stein T, Salomonis N, Nuyten DS, van de Vijver MJ, Gusterson BA. Stein T, Salomonis N, Nuyten DS, van de Vijver MJ, Gusterson BA. (2009) A mouse mammary gland involution mRNA signature identifies biological pathways potentially associated with breast cancer metastasis. J Mammary Gland Biol Neoplasia. 14(2):99-116.

Krishnan K, Salomonis N, Guo S. (2008) Identification of Spt5 target genes in zebrafish development reveals its dual activity in vivo. PLoS One. 3(11):e3621.

Salomonis N, Hanspers K, Zambon AC, Vranizan K, Lawlor SC, Dahlquist KD, Doniger SW, Stuart J, Conklin BR, Pico AR. (2007) GenMAPP 2: new features and resources for pathway analysis. BMC Bioinformatics. 24;8:217.

Cline MS, Blume J, Cawley S, Clark TA, Hu JS, Lu G, Salomonis N, Wang H, Williams A. (2005) ANOSVA: a statistical method for detecting splice variation from expression data. Bioinformatics. 21 Suppl 1:i107-i115.

Salomonis N, Cotte N, Zambon AC, Pollard KS, Vranizan K, Doniger SW, Dolganov G, Conklin BR. (2005) Identifying genetic networks underlying myometrial transition to labor. Genome Biol. 6:[R12](https://code.google.com/p/altanalyze/source/detail?r=12)–[R12](https://code.google.com/p/altanalyze/source/detail?r=12).16.

Doniger SW, Salomonis N, Dahlquist KD, Vranizan K, Lawlor SC, Conklin BR. (2003) MAPPFinder: Using Gene Ontology and GenMAPP to create a global gene-expression profile from microarray data. Genome Biol. 4:[R7](https://code.google.com/p/altanalyze/source/detail?r=7)–[R7](https://code.google.com/p/altanalyze/source/detail?r=7).12.

Dahlquist KD, Salomonis N, Vranizan K, Lawlor SC, and Conklin BR. (2002) GenMAPP: A new tool for viewing and analyzing microarray data on biological pathways. Nat. Genet. 31:19–20.

### RESEARCH SUPPORT ###
Active
ARRA Suppl. GM080223-06S1 (Conklin)
> Developing dynamic resources for GenMAPP-CS	08/01/09–07/31/11
As a supplement to the parent [R01](https://code.google.com/p/altanalyze/source/detail?r=01), this grant aims to add an overrepresentation analysis module (GO-Elite) and visualization module for alternative splicing data (AltAnalyze-CS) within the plugin architecture of Cytoscape and GenMAPP-CS.

[R01](https://code.google.com/p/altanalyze/source/detail?r=01) GM080223 (Conklin) GenMAPP-CS, a dynamic resource of pathway analysis	08/01/07–07/31/11

The major goal of this project is to develop and maintain GenMAPP-CS as a cross platform application tool that will provide the biological research community and the genomic databases in describing all of the pathways that make up a cell and how they interact as a system in the overall physiology of an organism.