### _What does the (+) and (-) mean proceeding a protein domain, miRNA binding site or protein identifier?_ ###

_**Answer**_: The (+) and (-) notations indicate which protein, domain or miRNA binding sites are predicted to predominate in the experimental group versus control. For example, (+)AA:997(ENSP00000311186)->1042(ENSP00000324892) indicates that the protein ENSP00000324892 should be increased relative to ENSP00000311186 in the experimental group (increases in size from 997 amino-acids to 1043 amino-acids). If this results was obtained from an exon array analysis for a probeset that was increased in an experiment relative to control (accounting for gene expression difference), then, one can conclude that the probeset aligns to the transcript that encodes for the longer protein (ENSP00000324892), whereas the alternative transcript does not align to this probeset (see the [AltAnalyze documentation](http://www.altanalyze.org/help.htm#inference) for details on protein assignment). However, if the probeset was downregulated, then the opposite is true (probeset aligns to the shorter isoform). If this annotation was proceeded by (-) rather than (+) and the probeset is upregulated, the probeset should align to ENSP00000311186 and indicates decreased expression of the longer isoform in the experimental relative to the control group. Thus, these annotations are most useful given the direction of the fold change of the alternative exon of interest. See the [following link](ScoringMethod.md) to see how to interpret alternative exon scores.

This same rationale is true for protein domains and microRNA binding site predictions. For a protein domain (e.g, kinase domain), a (+) most often indicates increased expression of the transcript region aligning to the domain in the experimental group versus the control. However, since a protein domain can be effected by inclusion or exclusion of an exon that is not directly overlapping with the domain, this domain can also exist up or downstream of the regulated exon. As a sanity check, you can always take the probeset sequence and BLAT align it a the http://genome.ucsc.edu and see which Ensembl transcripts it aligns to or even just view the data in DomainGraph. This should help the result make more sense.

A full description of fields for different examples can be found in the following:
  * [Splicing-Index results Excel table](http://altanalyze.org/image/SI_examples.xls)
  * [ASPIRE results Excel table](http://altanalyze.org/image/ASPIRE_examples.xls)
For more sophisticated comparisons, see the following paper:
http://dx.plos.org/10.1371/journal.pcbi.1000553